src/services/minecraft.service.ts

```diff
@@ -111,19 +116,31 @@ export class MinecraftService {
   }
 
   private async getScript(flavor: string, version: string): Promise<string> {
-    // GET HEAD, FLAVOR, AND TAIL PATHS
-    const headPath = resolve(__dirname, `../scripts/head.sh`);
-    const tailPath = resolve(__dirname, `../scripts/tail.sh`);
-    const flavorPath = resolve(__dirname, `../scripts/${flavor}.sh`);
-    // READ SCRIPT FILES
-    let headScript = await asyncReadFile(headPath, 'utf8');
-    const tailScript = await asyncReadFile(tailPath, 'utf8');
-    let flavorScript = await asyncReadFile(flavorPath, 'utf8');
+    // GET SCRIPT PATH
+    const scriptPath = resolve(__dirname, `../scripts/setup.sh`);
+
+    // READ SCRIPT FILE
+    let script = await asyncReadFile(scriptPath, 'utf8');
+
+    // FETCH URL BASED ON FLAVOR
+    let url = '';
+    if (flavor == 'vanilla') {
+      let vanilla = new Vanilla(version);
+      url = await vanilla.getServerUrl();
+    } else if (flavor == 'spigot') {
+      // spigot stuff
+    } else if (flavor == 'papermc') {
+      // papermc stuff
+    } else {
+      // throw error "invalid or unsupported flavor"
+    }
+
```

Comment:
@johnbwoodruff  This doesn't feel very elegant to me, but I'm not sure how to do something dynamically in Typescript (like creating a specific class based on a string that is passed in). Let me know if you have any ideas. I only added `spigot` and `papermc` to show my idea of how we could handle this section.

Reply:
Gave a suggestion, try that out and see if you prefer it. You could also use a `switch` statement if you think that looks more elegant.